src/Components/Aspire.Oracle.EntityFrameworkCore/AspireOracleEFCoreExtensions.cs

```diff
@@ -163,6 +164,20 @@ void ConfigureRetry()
 
     private static void ConfigureInstrumentation<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors | DynamicallyAccessedMemberTypes.NonPublicConstructors | DynamicallyAccessedMemberTypes.PublicProperties)] TContext>(IHostApplicationBuilder builder, OracleEntityFrameworkCoreSettings settings) where TContext : DbContext
     {
+        if (!settings.DisableTracing)
+        {
+            builder.Services.AddOpenTelemetry().WithTracing(tracerProviderBuilder =>
+            {
+                tracerProviderBuilder.AddOracleDataProviderInstrumentation(o =>
+                {
+                    o.EnableConnectionLevelAttributes = true;
+                    o.RecordException = true;
+                    o.InstrumentOracleDataReaderRead = true;
+                    o.SetDbStatementForText = true;
```

Comment:
This might be a silly question, but why aren't these set to true by default in `AddOracleDataProviderInstrumentation`? Are there other properties that should be set?

Reply:
One for the package authors as to why they're not the defaults. I did try originally with just an empty `AddOracleDataProviderInstrumentation()` constructor call but then didn't see any database calls being traced from the E2E playground. This is the same set of properties as used in the [Oracle OTel samples](https://github.com/oracle/dotnet-db-samples/blob/master/samples/opentelemetry/opentelemetry.cs).

Potentially it is because the Oracle OTel library isn't specifically aimed at EF so it might be the way that the Oracle EF package is setting the SQL statement, but hard to tell as it isn't OSS.

Reply:
Do you know any of the authors? Can you tag them here?

Reply:
Can't see Alex Keh on here, have raised https://github.com/oracle/dotnet-db-samples/issues/378 to ask

Reply:
@eerhardt @cmdkeen Oracle ODP.NET sets these properties to false by default as a security precaution. We didn't want developers to incorporate the ODP.NET OpenTelemetry and then inadvertently publish sensitive DB info into the trace. We want to make sure there was an explicit intent to do so.

Are other DB OpenTelemetry providers enabling these settings by default? I forget whether SqlClient has the options class members automatically enabled (and I couldn't locate the API doc).

Reply:
Hmm - I'm no expert, from looking at the playground output the Postgres and MySQL ones do emit at least some of this data (SQL command, passwordless connection string and db info) by default whereas it looks like the SqlServer one does not.
From my perspective having that level of detail is incredibly helpful. I can see at least four options:

1. Turning on this detail by default, mentioning in the docs
2. Switch back to the library defaults, allow fine grained control of the instrumentation options via OracleEntityFrameworkCoreSettings
3. Switch back to the library defaults, add a "DetailedTracing" property to OracleEntityFrameworkCoreSettings that sets these options with a suitable comment
4. Switch back to the library defaults - [not sure](https://docs.oracle.com/en/database/oracle/oracle-database/23/odpnt/OracleDataProviderInstrumentationOptionsClass.html) if you can change the options easily, otherwise users might need to disable tracing and manually add

I can't see any other examples of tracing configuration being passed through the constructors, and appreciate the extensions are trying to set up sensible defaults. My preference is for either 1 or 3 from above, but happy to take a steer.

Reply:
The intent of .NET Aspire components is to codify best practices. So developers will fall into a "pit of success" with using the defaults.

I don't think we should be enabling any setting by default if there are security concerns with that setting being on.